### PR TITLE
[WIP] fix: fixing the test: updated CI

### PR DESCRIPTION
**Pull request by anikethsaha**
Removed the old node version

Supporting node `v8.9.0`+

**WIP**
working on some test fix.

### CODE REVIEW

**Comment by anikethsaha** [comment]
Why this plugin is using older version of webpack (v3) even after updating (https://github.com/webpack-contrib/npm-install-webpack-plugin/pull/130/commits/278a7a9ac242df9c56232e6ae8efdee24e09d8c0)
 
image

@evilebottnawi

**Comment by evilebottnawi** [comment]
Need fix tests

**Comment by anikethsaha** [comment]
> Need fix tests

I dont know why its using old webpack version even after updating the webpack dependency

**Comment by evilebottnawi** [comment]
@anikethsaha need investigate

**Comment by evilebottnawi** [comment]
CI still failed, need fix

**Comment by anikethsaha** [comment]
Yup....doing that only

File: test/plugin.test.js [hunk 1]
```diff
@@ -1,51 +1,33 @@
...
-        loader: {
-          plugin: expect.createSpy(),
-          resolve: expect.createSpy(),
-        },
-        normal: {
-          plugin: expect.createSpy(),
-          resolve: expect.createSpy(),
-        },
-      },
-    };
```
Comment (anikethsaha) [inline test/plugin.test.js]: @evilebottnawi this was the problem. 
this test was passing the compiler object manually to the plugin.
now in webpack 4, the compiler instance is quite big and manually writing is not good.
So I used `webpack` to get the compiler, But the problem here is I cant put spies inside the its property cause its coming from webpack.

So I was thinking of doing non mocking/spying tests.
is it fine ?

**Comment by anikethsaha** [comment]
its still WIP

File: test/plugin.test.js [hunk 2]
```diff
@@ -56,136 +38,169 @@ describe("plugin", function() {
...
-  describe(".apply", function() {
-    it("should hook into `watch-run`", function() {
-      expect(this.compiler.plugin.calls.length).toBe(2);
-      expect(this.compiler.plugin.calls[0].arguments).toEqual([
-        "watch-run",
-        this.plugin.preCompile.bind(this.plugin)
-      ]);
+  describe('.apply', function() {
+    it('should hook into `watch-run`', function() {
+      expect(this.compiler.hooks.watchRun.taps.length).toBe(1);
```
Comment (anikethsaha) [inline test/plugin.test.js]: I meant doing these kind of test intead of spy/mock

**Pull request closed without merging**